app/liquid_tags/stackexchange_tag.rb

```diff
@@ -42,59 +46,54 @@ def render(_context)
 
   private
 
-  def valid_site?(site)
-    (site =~ /[a-z.]+/i)&.zero?
-  end
-
   def parse_site(input)
-    return "stackoverflow" if tag_name == "stackoverflow"
-
-    site = input.match(/[a-z.]+/i)[0]
-    unless valid_site?(site)
-      raise StandardError, I18n.t("liquid_tags.stackexchange_tag.invalid_site", tag: tag_name, input: input)
-    end
-
-    site
-  end
+    return "stackoverflow" if tag_name == "stackoverflow" || input.match?(STACKOVERFLOW_REGEXP)
 
-  def valid_input?(input)
-    return false if input.nil?
+    match = pattern_match_for(input, REGEXP_OPTIONS)
+    # rubocop:disable Layout/LineLength
+    raise StandardError, I18n.t("liquid_tags.stackexchange_tag.invalid_site", tag: tag_name, input: input) unless match && match_group_present?(match, "subdomain")
+    # rubocop:enable Layout/LineLength
 
-    ID_REGEXP.match?(input.split.first)
+    match[:subdomain].downcase
   end
 
-  def handle_response_error(response, input)
-    raise StandardError, "Calling StackExchange API failed: #{response&.error_message}" if response.code != 200
-
-    return unless response["items"].length.zero?
-
-    raise StandardError, I18n.t("liquid_tags.stackexchange_tag.post_not_found", tag: tag_name, input: input)
+  def match_group_present?(match, group_name)
+    match.names.include?(group_name)
   end
 
   def get_data(input)
-    unless valid_input?(input)
-      raise StandardError, I18n.t("liquid_tags.stackexchange_tag.invalid_id", tag: tag_name, input: input)
-    end
-
-    id = input.split.first
+    id, = input.split
+    match = pattern_match_for(id, REGEXP_OPTIONS)
+    # rubocop:disable Layout/LineLength
+    raise StandardError, I18n.t("liquid_tags.stackexchange_tag.invalid_id", tag: tag_name, input: input) unless match && match_group_present?(match, "id")
+    # rubocop:enable Layout/LineLength
```

Comment:
With the disabling and reenabling of the cop we use as many lines as a normal conditional would and the code would be easier to read since the important condition isn't on the far right:

```suggestion
    unless match && match_group_present?(match, "id")
      raise StandardError, I18n.t("liquid_tags.stackexchange_tag.invalid_id", tag: tag_name, input: input) 
    end
```